# แบบฝึกหัด 1

---

### 1.1

***เขียนโปรแกรมเพื่อเปลี่ยนค่าจากองศาเซลเซียสไปเป็นองศาฟาเรนไฮต์และเคลวิน ดังนี้***

F = (9/5)C + 32  
K = C + 273.15

รับข้อมูลองศาเซลเซียส

In [1]:
C = [39.85, 25, 32]

###แปลงเป็นองศาฟาเรนไฮต์ให้ได้ผลลัพธ์เป็น list เก็บใน F  

In [2]:
# Your code
def celsius_to_fahrenheit(celsius: float):
    return (celsius * 9/5) + 32

F = []
for c in C:
    F.append(celsius_to_fahrenheit(c))

print(F)

[103.73, 77.0, 89.6]


ผลลัพธ์ที่ต้องการแสดง [103.73, 77.0, 89.6]

###แปลงเป็นองศาฟาเคลวินให้ได้ผลลัพธ์เป็น list เก็บใน K

In [3]:
# Your code
def celsius_to_kelvin(celsius_list: [float]):
    return celsius_list + 273.15

K = []
for c in C:
    K.append(celsius_to_kelvin(c))

print(K)

[313.0, 298.15, 305.15]


ผลลัพธ์ที่ต้องการแสดง [313.0, 298.15, 305.15]

###แสดงการสรุปผล

In [4]:
# Your code
#

# ตัวอย่างเพื่อให้ print °C
x = 10
print(f"{x}°C")

10°C


ผลที่ต้องการแสดง  
39.85 °C = 103.73 °F = 313.0 °K  
25 °C = 77.0 °F = 298.15 °K  
32 °C = 89.6 °F = 305.15 °K  

---

### 1.2

ให้นิสิตแสดงวิธีการแปลง list income (ซึ่งเปรียบเสมือนเป็นข้อมูลเงินเดือนของผู้ตอบแบบสอบถาม 20 ท่าน)  
ให้เป็น numpy array เก็บในตัวแปรที่ชื่อ income_number โดย array ที่แปลงได้ควรมีชนิด dtype เป็น float   
การแปลง string ให้เป็น float กำหนดให้ K = 1000 และ M = 1000000   
string "-" ให้แปลงเป็น np.nan (dtype ต้องเป็น float จึงจะใช้ np.nan ได้)

\*โจทย์ข้อนี้สามารถทำได้หลายวิธี  
หากใช้ numpy นิสิตอาจต้องหาข้อมูลเพิ่ม เช่น การใช้ np.char.replace หรือฟังก์ชัน eval

In [5]:
import numpy as np

In [6]:
income = ["20K", "3M", "12K", "0.04M", "0.023K", "-", "16K", "740K", "15K", "40K",
         "-", "17K", "-", "120K", "56K", "-", "42K", "2M", "0.14M", "60K"]

In [7]:
eval("2*1000")

2000

In [8]:
income_number = np.array([])

# นิสิตแสดงวิธีการด้วยการเขียนโค้ด


##วิธีการของนิสิตควรได้ผลลัพธ์ดังที่ระบุเมื่อรัน cell ต่อไปนี้

In [9]:
print(income_number)

[]


ผลลัพธ์ที่ควรแสดง: array([2.0e+04, 3.0e+06, 1.2e+04, 4.0e+04, 2.3e+01,     nan, 1.6e+04,
       7.4e+05, 1.5e+04, 4.0e+04,     nan, 1.7e+04,     nan, 1.2e+05,
       5.6e+04,     nan, 4.2e+04, 2.0e+06, 1.4e+05, 6.0e+04])

In [10]:
type(income_number)

numpy.ndarray

ผลลัพธ์ที่ควรแสดง: numpy.ndarray

In [11]:
income_number.dtype

dtype('float64')

ผลลัพธ์ที่ควรแสดง: dtype('float64')

In [12]:
np.nanmean(income_number)

C:\Users\User\AppData\Local\Temp\ipykernel_18804\1254672693.py:1: RuntimeWarning: Mean of empty slice
  np.nanmean(income_number)


np.float64(nan)

ผลลัพธ์ที่ควรแสดง: 394876.4375

In [13]:
np.nanmedian(income_number)

c:\Users\User\AppData\Local\Programs\Python\Python313\Lib\site-packages\numpy\lib\_nanfunctions_impl.py:1215: RuntimeWarning: Mean of empty slice
  return np.nanmean(a, axis, out=out, keepdims=keepdims)


np.float64(nan)

ผลลัพธ์ที่ควรแสดง: 41000.0

---